### PR TITLE
Support Etag

### PR DESCRIPTION
**Pull request by n0rad**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -29,6 +29,10 @@ import (
 	"text/template"
 	"time"
 
+	"crypto/sha1"
```
Comment (lucab) [inline api.go]: Nit: spurious newline before this, preventing proper gofmt import ordering.

File: api.go [hunk 2]
```diff
@@ -88,6 +91,42 @@ func (a ACApi) Start() {
...
+}
+
+func (a ACApi) fileHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !strings.HasPrefix(r.URL.Path, "/") {
+			r.URL.Path = "/" + r.URL.Path
+		}
+
+		file := a.storage.RootPath + path.Clean(r.URL.Path)
+		if _, err := os.Stat(file); !os.IsNotExist(err) {
```
Comment (lucab) [inline api.go]: This seems to be missing a `err != nil && `.

**Comment by n0rad** [comment]
fixed

**Review by lucab — APPROVED** [review]
LGTM (as a casual bystander)

**Pull request merged**